ingress-node-firewall/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols
+type IngressNodeFirewallICMPRule struct {
+	// ICMPType define ICMP Type Numbers (RFC 792).
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=255
+	// +kubebuilder:validation:Minimum:=0
+	// +optional
+	ICMPType uint8 `json:"icmpType,omitempty"`
+
+	// ICMPCode define ICMP Code ID (RFC 792).
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=16
+	// +kubebuilder:validation:Minimum:=0
+	// +optional
+	ICMPCode uint8 `json:"icmpCode,omitempty"`
+}
+
+// IngressNodeFirewallProtoRule define ingress node firewall rule for TCP, UDP and SCTP protocols
+type IngressNodeFirewallProtoRule struct {
+	//Ports can be a single port or range of ports using start-end range format
+	//+kubebuilder:validation:Optional
+	// +optional
+	Ports string `json:"ports,omitempty"`
+}
+
+// IngressNodeFirewallProtocolRule define ingress node firewall rule per protocol
+type IngressNodeFirewallProtocolRule struct {
+	// Order define order of execution of ingress firewall rules .
```

Comment:
What happens if multiple rules specify the same order number?

Reply:
validation webhook detect and error the config

Reply:
If you made this a `+listType=map`, `+listMapKey=order` then the user wouldn't be able to do this and the OpenAPI schema would reject it, you then wouldn't need a webhook I believe